pkg/apis/pipeline/v1beta1/customrun_types.go

```diff
@@ -0,0 +1,246 @@
+
+	// Used for cancelling a customrun (and maybe more later on)
+	// +optional
+	Status CustomRunSpecStatus `json:"status,omitempty"`
+
+	// Status message for cancellation.
+	// +optional
+	StatusMessage CustomRunSpecStatusMessage `json:"statusMessage,omitempty"`
+
+	// Used for propagating retries count to custom tasks
+	// +optional
+	Retries int `json:"retries,omitempty"`
+
+	// +optional
+	ServiceAccountName string `json:"serviceAccountName"`
+
+	// Time after which the custom-task times out.
+	// Refer Go's ParseDuration documentation for expected format: https://golang.org/pkg/time/#ParseDuration
+	// +optional
+	Timeout *metav1.Duration `json:"timeout,omitempty"`
+
+	// Workspaces is a list of WorkspaceBindings from volumes to workspaces.
+	// +optional
+	// +listType=atomic
+	Workspaces []WorkspaceBinding `json:"workspaces,omitempty"`
+}
+
+// CustomRunSpecStatus defines the taskrun spec status the user can provide
+type CustomRunSpecStatus string
+
+const (
+	// CustomRunSpecStatusCancelled indicates that the user wants to cancel the run,
+	// if not already cancelled or terminated
+	CustomRunSpecStatusCancelled CustomRunSpecStatus = "RunCancelled"
+)
+
+// CustomRunSpecStatusMessage defines human readable status messages for the TaskRun.
+type CustomRunSpecStatusMessage string
+
+const (
+	// CustomRunCancelledByPipelineMsg indicates that the PipelineRun of which part this CustomRun was
+	// has been cancelled.
+	CustomRunCancelledByPipelineMsg CustomRunSpecStatusMessage = "CustomRun cancelled as the PipelineRun it belongs to has been cancelled."
+	// CustomRunCancelledByPipelineTimeoutMsg indicates that the Run was cancelled because the PipelineRun running it timed out.
+	CustomRunCancelledByPipelineTimeoutMsg CustomRunSpecStatusMessage = "CustomRun cancelled as the PipelineRun it belongs to has timed out."
+)
+
+// GetParam gets the Param from the CustomRunSpec with the given name
+// TODO(jasonhall): Move this to a Params type so other code can use it?
+func (rs CustomRunSpec) GetParam(name string) *Param {
+	for _, p := range rs.Params {
+		if p.Name == name {
+			return &p
+		}
+	}
+	return nil
+}
+
+const (
+	// CustomRunReasonCancelled must be used in the Condition Reason to indicate that a CustomRun was cancelled.
+	CustomRunReasonCancelled = "CustomRunCancelled"
+	// CustomRunReasonTimedOut must be used in the Condition Reason to indicate that a CustomRun was timed out.
+	CustomRunReasonTimedOut = "CustomRunTimedOut"
+	// CustomRunReasonWorkspaceNotSupported can be used in the Condition Reason to indicate that the
+	// CustomRun contains a workspace which is not supported by this custom task.
+	CustomRunReasonWorkspaceNotSupported = "CustomRunWorkspaceNotSupported"
+	// CustomRunReasonPodTemplateNotSupported can be used in the Condition Reason to indicate that the
+	// CustomRun contains a pod template which is not supported by this custom task.
+	CustomRunReasonPodTemplateNotSupported = "CustomRunPodTemplateNotSupported"
```

Comment:
if we're removing pod template this can be removed as well

Reply:
done, thank you!